webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -216,11 +218,44 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
       };
       this.contactId = null;
       formId = contactType.create_form;
+
+      await this.validateParentForCreateForm();
     }
 
     return formId;
   }
 
+  private async validateParentForCreateForm() {
+    const datasource = await this.chtDatasourceService.get();
+
+    if (!this.contact.parent) {
+      const topLevelTypes = await this.contactTypesService.getChildren();
+      if (!topLevelTypes.some(({ id }) => id === this.contact.contact_type)) {
+        throw new Error(`Cannot create a ${this.contact.contact_type} at the top level. It requires a parent.`);
+      }
+      return;
+    }
+
+    const parent = await datasource.v1.place.getByUuid(this.contact.parent);
+    if (!parent) {
+      throw new Error(`Parent with UUID ${this.contact.parent} does not exist.`);
+    }
+
+    if (await this.contactTypesService.isPerson(parent)) {
+      throw new Error(`Cannot create a ${this.contact.contact_type} as a child of a person.`);
+    }
```

Comment:
Sorry for going back and forth on this. :facepalm: Seeing this `isPerson` check got me to wondering and then digging/testing.  It appears that we actually do not have anything currently preventing a configuration where `person` contacts are parents of other types of contacts.  I have logged https://github.com/medic/cht-core/issues/9584 seeking more clarity on this issue for the future, but for now, I don't think we want to take a strong position in this code.

Instead, I think we should switch to just use the `dbService` instead of the `datasource` to get the parent (since the `datasource` does not yet support just getting generic "contacts").  One "advantage" to this approach is that Pouch should throw an error if it cannot find the parent. That means we can eliminate these two if-blocks and make Sonar happy!  :+1: 

```suggestion
    const parent = await this.dbService
      .get()
      .get(this.contact.parent);
```